header.php

```diff
@@ -1,385 +1,385 @@
+				}
+			  }
+			  xmlhttp.open("GET","/beatmapSearch.php?q="+str,true);
+			  xmlhttp.send();
+			}
+			
+			function searchFocus() {
+				document.getElementById("topBarSearchResults").style.display="block";
+			}
+		</script>
+	</head>
+	<body>
+		<div class="topBar">
+			<a href="/" style="margin-right: 8px;">OMDB - osu! map database</a>
+			
+			<a href="/"><div class="topBarLink">home</div></a>
+			<a href="/charts/"><div class="topBarLink">charts</div></a>
+			<div class="topBarDropDown">
+				<div class="topBarLink topBarDropDownButton">maps</div>
+				<div class="dropdown-content">
+					<a href="/maps/?m=01&y=2023">latest</a>
+					<a href="/random/">random</a>
+				</div>
+			</div>
+			
+			<form class="topBarSearch">
+				<input class="topBarSearchBar" type="text" size="30" onfocusin="searchFocus()" onkeyup="showResult(this.value)" value="" autocomplete="off" placeholder="Search... (or paste link)">
+				<div id="topBarSearchResults"></div>
+			</form>
+			
+			<?php
+				function FetchOsuOauthLink($oauthClientID, $serverProtocol) {
+					// the creation of the local function scope in php was a disaster for humanity -t
+					$oauthFields = array(
+						"client_id" => $oauthClientID,
+						"redirect_uri" => $serverProtocol . $_SERVER['SERVER_NAME'] . '/callback.php',
+						"response_type" => "code",
+						"scope" => "identify public",
+						"state" => "z"
+					);
+					return 'https://osu.ppy.sh/oauth/authorize?' . http_build_query($oauthFields);
+				}
+			?>
 
-            <span style="float:right;">
-                <?php
-                    if ($loggedIn) {
-                ?>
-                    <a href="/profile/<?php echo $userId; ?>" style="color:white;"><img src="https://s.ppy.sh/a/<?php echo $userId; ?>" style="height:2rem;vertical-align:middle;">&ZeroWidthSpace;</img></a> <a href="/profile/<?php echo $userId; ?>" style="color:white;"><b><?php echo $userName; ?></b></a>
-                <?php
-                    } else {
-                        include_once 'sensitiveStrings.php'; // needed for $clientID
-                        $oauthProtocol = isset($_SERVER['HTTPS']) ? 'https://' : 'http://';
-                ?>
-                    <b><a href=<?php echo FetchOsuOauthLink($clientID, $oauthProtocol); ?> style="color:white;">log in</a></b>
-                <?php
-                    }
-                ?>
-            </span>
-        </div>
+			<span style="float:right;">
+				<?php
+					if ($loggedIn) {
+				?>
+					<a href="/profile/<?php echo $userId; ?>" style="color:white;"><img src="https://s.ppy.sh/a/<?php echo $userId; ?>" style="height:2rem;vertical-align:middle;">&ZeroWidthSpace;</img></a> <a href="/profile/<?php echo $userId; ?>" style="color:white;"><b><?php echo $userName; ?></b></a>
+				<?php
+					} else {
+						include_once 'sensitiveStrings.php'; // needed for $clientID
+						$oauthProtocol = isset($_SERVER['HTTPS']) ? 'https://' : 'http://';
```

Comment:
will probably move this 1 to `functions.php` later or whatever is used for constants, cuz rn oauth is hardcoded to ur instance anyway and i might fix that up later